### DISCUSSION TITLE
Hide/Show transition on accordion

### DISCUSSION THREAD

**Discussion by DicksB** [tabpanelwidget/tabpanelwidget#6] [Ideas]
Have you considered making it possible to add a hide/show transition on the .tpw-panel content when the accordion opens and closes? I realize it can't go on the panel itself, since the shim is where the hide/show is actually happening, and the documentation warns against styling tpw-shim.

For us, the use case is simply that we need to support both TPW and accordions created by the CKEditor Accordion add-on, and we're trying to make their appearance and behavior match as closely as possible without interfering with TPW's accessibility advantages.

**Comment by thierryk** [comment]
I see a few issues trying to make TPW create transitioning effects. I'm listing them here in case you are trying to customize the widget on your own and are not sure what you could "break":

   * we are hiding/showing the panels using `display:none/inline`. This is a styling "trick" that allows us to use sequential headers/panels instead of having to use "jump links" like what most (all?) TabPanels do out there. Unfortunately, simply "transitioning" from a hidden element (styled with `display:none`) is not possible.
   * we could hide the panels using something else than `display:none` (as `display:none` is only **needed** for the TabPanel really) when styling the widget as an Accordion (to be able to do some transition) but I'm afraid that could create some issues with the interface when the browser dynamically switches from Accordion to TabPanel and vice-versa. 
   * hiding the panels only visually (for Accordions) may solve the styling issue but that will create problems with Screen-Reader users who will have access to all ARIA attributes related to a Accordion widget (i.e. `aria-controls` and `role`) even though everything will be accessible to them (no panels would be hidden from them).

One thing we could try is to apply a class onto the panel a few milliseconds after it is revealed. This should allow some add/show transition/styling but could also create weird behavior related to "timing". I'm not sure...

**Reply by thierryk** [reply]
Applying a class would help trigger the transition when the panel is revealed but how can we take care of the inverse effect, when the panel is hidden again? This would require to remove the class before "closing" the panel (with `display:none`), making sure that that "delay" matches the delay of the transition... :-(

**Reply by DicksB** [reply]
I appreciate you thinking about this! It does sound like it may be more trouble than it's worth. We may be better off overriding the styling on CKEditor's accordions to disable the transitions there.

If you're interested, this came up while working on a [Drupal sandbox project](https://www.drupal.org/sandbox/wrd/3224554). I haven't decided yet if I'll make it a full project (not sure I have the time to support it), but I'm pretty pleased with how it's working so far.

**Reply by thierryk** [reply]
> It does sound like it may be more trouble than it's worth

I'm afraid it is...

> If you're interested, this came up while working on a [Drupal sandbox project](https://www.drupal.org/sandbox/wrd/3224554).

Good luck with this! Please do not hesitate to come back here with questions if you run into any issue.
